packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -77,16 +100,42 @@ export async function runtimeWalkthrough(
     },
   };
   const contribution = await plugin.contribute(contributionRequest);
-  return {
-    ...contribution,
-    runtimePluginId: selection.pluginId,
-  };
+  return [
+    {
+      ...contribution,
+      runtimePluginId: selection.pluginId,
+    },
+  ];
+}
+
+async function _layerRuntimeWalkthroughHelper(
+  params: Partial<LayerParameters>,
+  plugin,
+  selections,
+): Promise<Array<Pick<FunctionParameters, 'runtimePluginId'> & FunctionRuntimeParameters>> {
+  const retVal = [];
+  for (let selection of selections) {
+    const contributionRequest: ContributionRequest = {
+      selection: selection.value,
+      contributionContext: {
+        runtime: params.runtimes[0],
+        functionName: params.layerName,
+        resourceName: params.layerName,
+      },
+    };
+    const contribution = await plugin.contribute(contributionRequest);
```

Comment:
why are we asking the same plugin for a runtime multiple times? Each plugin (currently) can only supply one runtime so you're going to get the same thing back every time

Reply:
fixed